crates/ra_ide/src/imports_locator.rs

```diff
@@ -0,0 +1,97 @@
+//! This module contains an import search funcionality that is provided to the ra_assists module.
+//! Later, this should be moved away to a separate crate that is accessible from the ra_assists module.
+
+use crate::{
+    db::RootDatabase,
+    references::{classify_name, classify_name_ref, NameDefinition, NameKind},
+    symbol_index::{self, FileSymbol},
+    Query,
+};
+use ast::NameRef;
+use hir::{db::HirDatabase, InFile, ModPath, Module, SourceBinder};
+use itertools::Itertools;
+use ra_assists::ImportsLocator;
+use ra_prof::profile;
+use ra_syntax::{ast, AstNode, SyntaxKind::NAME};
+
+pub(crate) struct ImportsLocatorIde<'a> {
+    source_binder: SourceBinder<'a, RootDatabase>,
+}
+
+impl<'a> ImportsLocatorIde<'a> {
+    pub(crate) fn new(db: &'a RootDatabase) -> Self {
+        Self { source_binder: SourceBinder::new(db) }
+    }
+
+    fn search_for_imports(
+        &mut self,
+        name_to_import: &ast::NameRef,
+        module_with_name_to_import: Module,
+    ) -> Vec<ModPath> {
+        let _p = profile("search_for_imports");
+        let db = self.source_binder.db;
+        let name_to_import = name_to_import.text();
+
+        let project_results = {
+            let mut query = Query::new(name_to_import.to_string());
+            query.exact();
+            query.limit(10);
```

Comment:
It feels like there better be some limit, but what should it be?

Reply:
It might be better to choose the limit here a bit higher and limit the actual choices more afterwards...

Reply:
Why not, not sure how to properly limit it , so have put 40 there for each query and added `.take(20)` right before returning the results.